caf: Add ble_adv_data_update_event (#8507)
nrfconnect/sdk-nrf: doc/nrf/libraries/caf/ble_adv.rst

The |ble_adv| uses Zephyr's :ref:`zephyr:settings_api` to store the information if the peer for the given local identity uses the Resolvable Private Address (RPA).

Undirected advertising data update
==================================

> The module does not instantly update advertising and scan response payloads when advertising data or scan response data provided by :ref:`bt_le_adv_prov_readme` is modified.

Reviewer: ```suggestion
The module does not instantly update advertising and scan response payloads when either advertising data or scan response data (provided by :ref:`bt_le_adv_prov_readme`) is modified.
```

> The module automatically gets new advertising data and scan response data from Bluetooth LE advertising data proviers subsystem only in the following cases:

Reviewer: ```suggestion
The module automatically gets new advertising data and scan response data from Bluetooth LE's advertising data provider subsystem only in the following cases:
```

* Bluetooth LE undirected advertising is started or restarted.

> * Undirected advertising enters grace period.

Reviewer: ```suggestion
* Undirected advertising enters the grace period.
```
Plus link to the grace period section, no?

Author: Ok, I will add it (then we need to merge the https://github.com/nrfconnect/sdk-nrf/pull/8457 first

> The payload update can be triggered by an application using :c:struct:`ble_adv_data_update_event`.

Reviewer: ```suggestion
The payload update can be triggered by the application using :c:struct:`ble_adv_data_update_event`.
```

> Make sure to submit the event after changing Bluetooth data provided by a provider.

Reviewer: ```suggestion
Make sure to submit the event after changing the Bluetooth data provided by a provider.
```
